WordPress/gutenberg
# Update filter control position

## What?
This PR moves the data view filter controls into the same row as the search, bulk actions, and view options controls.

## Why?
When no (or few) filters are set (which is most common), two separate rows for these controls doesn't feel like the best use of space. 

The trade-off is that on smaller screens the controls move around when you begin adding lots of filters. However, for 6.5 core data views have at most two filters, so this feels a bit of an edge case. In the future filter controls may permanently require a dedicated row, but I'm not sure that's the case right now.

## Testing Instructions
1. Open a data view in the site editor e.g. "Manage all pages"
2. Notice the "Add filter" and "Reset filters" buttons appear adjacent to the search input
3. Adds filters
4. Observe the the filter controls wrap, as demonstrated in the video below.



https://github.com/WordPress/gutenberg/assets/846565/3953c63f-c96a-4963-9270-e01c67e02016
